### ISSUE TITLE
[Bug]: NVIDIA Tesla P40 Card Issues

### ISSUE THREAD

**Issue by nphil** [Steam-Headless/docker-steam-headless#161]
### Describe the Bug

Apparently there's a problem with the newer Nvidia drivers and this Pascal based workstation card. They're available pretty cheap on Ebay (100-150USD) and are excellent for AI tasks with 24GB of VRAM so I'm assuming these will be fairly popular for homelabbers.

These cards do not have any kind of display out capability. Regardless, I was able to run games just fine until recently  - Steam goes haywire and Sunshine doesn't work at all (no display detected). See below screenshot for Webui:

![image](https://github.com/user-attachments/assets/12fa5b2b-7c5f-46cf-9773-b65cefbbd2f4)

Mikec92117 on discord advised that the last working driver version was 535.183.01, however I do not see an option to download this on unRAID with the Nvidia plugin.

Is it possible to manually select this driver through steam-headless or ideally, what could be done to fix the issue on the latest NVIDIA drivers?

PS: The latest driver works fine for compute tasks, I use Ollama, Plex transcoding etc. and everything works.

### Screenshots

![image](https://github.com/user-attachments/assets/03929b2c-7e4a-456c-9c93-337168e90673)

### Version

steam-headless:latest

### Platform

Slackware - 15.0
6.1.99-Unraid
NVIDIA-SMI 550.107.02

**Comment by karl0ss** [comment]
I see the same on my Nvidia tesla m60

Can see in `nvidia-smi` that steam is using G type and appears to be using the card, but blank windows as you see

**Comment by nphil** [comment]
I went back to driver 470.x via unRAID and everything works fine. Would rather figure out the issue though as I really want to keep the drivers updated.

**Comment by karl0ss** [comment]
so you had to pull the drivers back on your actual host? thats a bit of a shame...hmmm

**Comment by nphil** [comment]
yup, but i was wondering if there was a way to make the docker image pull the older version of the driver during first run - should technically able to modify the script but it's beyond my ability.

**Comment by karl0ss** [comment]
I can confirm that rolling my host back to `535.183.01` has fixed this for my Tesla M60 :)

**Comment by aa889788** [comment]
could you try Forza Horizon 4 or RDR2 in DX12 mode? My Tesla P4 failed to launch these games in DX12 mode. Is this a common issue of Pascal card?

**Comment by karl0ss** [comment]
@aa889788 how can we play Forza? can we install the xbox for windows app or something?

**Comment by aa889788** [comment]
> @aa889788 how can we play Forza? can we install the xbox for windows app or something?

The steam version, but I've just fixed it by delete the `TargetHardwareProfiler.dll`

**Comment by eivanov-c** [comment]
Hi.
I have the same error.
<img width="1606" alt="image" src="https://github.com/user-attachments/assets/61230ba6-19b3-46e3-ac25-2469b3e371b0">
Host is arch linux and in host nvidia-smi see the processes
<img width="720" alt="image" src="https://github.com/user-attachments/assets/d5cc1bcb-7f9e-48ac-b4d1-5ba4ba4da10d">
Could you help to understand what's wrong?

**Comment by karl0ss** [comment]
Not sure, but that is what I saw on the incompatible driver, I know for sure `535.183.01` worked, as that is what I downgraded to and it all started wroking...

things like sunshine should show as G+C for the type when its working...

**Comment by eivanov-c** [comment]
I have tried several versions of drivers
nvidia
nvidia-dkms
nvidia-470
nvidia-535.183.01
but each time I have the same error

**Comment by nphil** [comment]
For those using the official unraid container from Community apps:

If anyone else is trying to set an old driver version manually via an environment variable, you must remove the --runtime='nvidia' from your extra parameters and replace with --device='/dev/dri' instead. You will then need to add an extra variable NVIDIA_DRIVER_VERSION and give it a value of 535.183.01. 

Everything should work now as the container will download the older driver version instead of the latest.

**Comment by Josh5** [comment]
Closing as superseded by the imminent Steam Headless V2 release, expected in the coming weeks. V2 does not require a physical display output, but the reported NVIDIA 550 driver is below the current Sunshine requirement; please open a fresh V2 issue on a supported driver if needed.
